### PR TITLE
feat(spans): Keep domain + extension for image resources

### PR DESCRIPTION
**Pull request by jjbayer**
Group `resource.img` spans by domain + extension. Known path segments like `data` are also allowed.

This PR does not yet enable span ingestion for all orgs (still restricted to `all-modules`).

ref: https://github.com/getsentry/sentry/issues/61161

### CODE REVIEW

File: relay-event-normalization/src/normalize/span/description/mod.rs [hunk 1]
```diff
@@ -298,14 +301,20 @@ fn scrub_resource_filename(path: &str) -> String {
...
-    let mut joined = segments.join("/");
-    if !extension.is_empty() {
-        joined.push('.');
-        joined.push_str(extension);
-    }
+    let basename = if ty == "img" {
+        Cow::Borrowed("*")
+    } else {
+        scrub_resource_segment(basename)
+    };
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/description/mod.rs:308]: This is the main functional change in this PR: for `img` spans, only keep the extension.

File: relay-event-normalization/src/normalize/span/description/mod.rs [hunk 2]
```diff
@@ -298,14 +301,20 @@ fn scrub_resource_filename(path: &str) -> String {
         extension = "";
     }
 
-    let mut segments = base_path.split('/').map(scrub_resource_segment);
-    let mut joined = segments.join("/");
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/description/mod.rs:302]: Drive-by fix: Segments are already handled in `scrub_resource`, so no need to do it again here.

File: relay-event-normalization/src/normalize/span/description/mod.rs [hunk 3]
```diff
@@ -553,16 +562,30 @@ mod tests {
...
         "resource.img",
-        "https://*.domain.com/data/guide.gif"
+        "https://*.domain.com/data/*.gif"
+    );
+
+    span_description_test!(
+        resource_no_ids_img,
+        "https://data.domain.com/something/guide.gif",
+        "resource.img",
+        "https://*.domain.com/*/*.gif"
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/description/mod.rs:595]: @DominikB2014 these tests illustrate what the new span descriptions will look like.

**Review by olksdr — APPROVED** [review]

File: relay-event-normalization/src/normalize/span/description/mod.rs [hunk 4]
```diff
@@ -723,15 +746,15 @@ mod tests {
 
     span_description_test!(
         resource_img_comma_with_extension,
-        "https://example.org/p/fit=cover,width=150,height=150,format=auto,quality=90/media/photosV2/weird-stuff-123-234-456.jpg",
-        "resource.img",
-        "https://example.org/*/media/*/weird-stuff-*-*-*.jpg"
+        "https://example.org/p/fit=cover,width=150,height=150,format=auto,quality=90/media/photosV2/weird-stuff-123-234-456.js",
+        "resource.script",
```
Comment (iker-barriocanal) [inline relay-event-normalization/src/normalize/span/description/mod.rs:771]: Why are we changing existing tests, instead of adding new ones? Should we update the test names if we update what they're testing?

**Review by iker-barriocanal — APPROVED** [review]

File: relay-event-normalization/src/normalize/span/description/mod.rs (hunk 4 above)
Comment (jjbayer) [reply relay-event-normalization/src/normalize/span/description/mod.rs:771]: The intention of the test was to make sure commas don't make it into the scrubbed description, and that is now trivially true for `resource.img` because the entire base name gets removed. But you are right, I will duplicate the tests instead.

**Review by DominikB2014 — APPROVED** [review]

**Comment by DominikB2014** [comment]
One thing we might have to consider in the future is that we can still have images that don't have the span.op:resource.img, once merged, I can monitor how it looks in this case.
